### PR TITLE
tx lookup File Write (hack.go)

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: cmd/hack/hack.go [hunk 1]
```diff
@@ -1666,6 +1681,243 @@ func GenerateTxLookups1(chaindata string, block int) {
...
+	}
+
+	for !interrupt && len(*h) != 0 {
+		val := (heap.Pop(h)).(HeapElem)
+		if lookups[val.index].pos == uint64(bufferLen) {
+			if val.val[32] != 0 {
+				err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[32:]))
+				check(err)
+			} else {
+				err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[33:]))
```
Comment (AlexeyAkhunov) [inline cmd/hack/hack.go]: Why is it `val.val[33:]` here and `val.val[32:]` up there? Is this something to do with the encoding of block numbers?

File: cmd/hack/hack.go [hunk 2]
```diff
@@ -1666,6 +1681,243 @@ func GenerateTxLookups1(chaindata string, block int) {
...
+				check(err)
+			}
+			n, _ := lookups[val.index].file.Read(lookups[val.index].buffer)
+			iterations++
+			if n == 0 {
+				os.Remove(fmt.Sprintf(".lookups_%d.tmp", val.index))
```
Comment (AlexeyAkhunov) [inline cmd/hack/hack.go]: Don't you need to close the file before removing it?

File: cmd/hack/hack.go (hunk 1 above)
Comment (Giulio2002) [reply cmd/hack/hack.go]: yes, in case of trailing 0s

File: cmd/hack/hack.go [hunk 3]
```diff
@@ -1666,6 +1681,243 @@ func GenerateTxLookups1(chaindata string, block int) {
...
+			} else {
+				if n != bufferLen {
+					lookups[val.index].buffer = lookups[val.index].buffer[:n]
+				}
+				lookups[val.index].pos = 35
+				heap.Push(h, HeapElem{lookups[val.index].buffer[:35], val.index})
+			}
+			continue
+		} else {
```
Comment (AlexeyAkhunov) [inline cmd/hack/hack.go]: You can remove this `else` branch, because corresponding `if` always does `continue`

File: cmd/hack/hack.go [hunk 4]
```diff
@@ -1666,6 +1681,250 @@ func GenerateTxLookups1(chaindata string, block int) {
...
+		default:
+		}
+		blockBytes.SetUint64(blockNum)
+		bn = blockBytes.Bytes()
+
+		for _, tx := range body.Transactions {
+			val, err := db.Get(dbutils.TxLookupPrefix, tx.Hash().Bytes())
+			iterations++
+			if iterations%100000 == 0 {
+				batch.Commit()
```
Comment (AlexeyAkhunov) [inline cmd/hack/hack.go]: You probably don't want this commit here. It slows down the validation, I suspect

**Pull request merged**